book/run/optimism.md

```diff
@@ -0,0 +1,121 @@
+
+**`rethdb` build tag**  
+The `op-node` also comes with an experimental `rethdb` build tag, which allows it to read receipts directly from an L1 `reth` database during [derivation][derivation-spec]. This can speed up sync times, but it is not required if you do not
+have access to the L1 archive node on the same machine as your L2 node.
+
+To build the `op-node` with the `rethdb` build tag enabled:
+```sh
+git clone git@github.com:ethereum-optimism/optimism.git && \
+    (cd optimism/op-service/rethdb-reader && cargo build --release) && \ 
+    cd optimism/op-node && \
+    go build -v -tags rethdb -o ./bin/op-node ./cmd/main.go && \
+    mv bin/op-node /usr/bin/op-node
+```
+This will build the `rethdb-reader` dylib and instruct the `op-node` build to statically link this dylib into the binary. The `op-node` binary will be installed to `/usr/bin/op-node`.
+
+### Running `op-reth`
+
+The `optimism` feature flag in `op-reth` adds several new CLI flags to the `reth` binary:
+1. `--rollup.sequencer-http <uri>` - The sequencer endpoint to connect to. Transactions sent to the `op-reth` EL are also forwarded to this sequencer endpoint for inclusion, as the sequencer is the entity that builds blocks on OP Stack chains.
+1. `--rollup.disable-tx-pool-gossip` - Disables gossiping of transactions in the mempool to peers. This can be ommitted for personal nodes, though providers should always opt to enable this flag.
+1. `--rollup.enable-genesis-walkback` - Disables setting the forkchoice status to tip on startup, making the `op-node` walk back to genesis and verify the integrity of the chain before starting to sync. This can be ommitted unless a corruption of local chainstate is suspected.
+
+Base's `rollup.json` files, which contain various configuration fields for the rollup, can be found in their [node][base-node] repository, under the respective L1 settlement layer's directory (`mainnet`, `goerli`, & `sepolia`).
+
+First, ensure that your L1 archival node is running and synced to tip. Then, start `op-reth` with the `--rollup.sequencer-http` flag set to the `Base Mainnet` sequencer endpoint:
+```sh
+op-reth node \
+    --chain base \
+    --rollup.sequencer-http https://sequencer.base.org \
+    --http \
+    --ws \
+    --authrpc.port 9551 \
+    --authrpc.jwtsecret /path/to/jwt.hex
+```
+
+Then, once `op-reth` has been started, start up the `op-node`:
+```sh
+op-node \
+    --l1=<your-L1-rpc> \
+    --l2=http://localhost:9551 \
+    --rollup.config=/path/to/rollup.json \
+    --l2.jwt-secret=/path/to/jwt.hex \
+    --rpc.addr=0.0.0.0 \
+    --rpc.port=7000 \
+    --l1.trustrpc
+```
+
+If you opted to build the `op-node` with the `rethdb` build tag, this "`RPCKind`" can be enabled via appending two extra flags to the `op-node` invocation:
+```sh
+op-node \
+    # ...
+    --l1.rpckind=reth_db \
+    --l1.rethdb=<your_L1_reth_db_path>
```

Comment:
is this a reth datadir, `db` dir inside it or `mdbx.dat` file path?

Reply:
Added a clarification

Reply:
I think doing `--l1.rethdb=$(reth db path)` would be confusing, but it's the easiest way to get the path.

Reply:
Done 👍

Reply:
sick